lighthouse-viewer/test/viewer-test-pptr.js

```diff
@@ -59,80 +65,167 @@ describe('Lighthouse Viewer', function() {
-        const audit = el.closest(selectors.audits);
-        const auditTitle = audit && audit.querySelector(selectors.titles);
-        return {
-          explanation: el.textContent,
-          title: auditTitle ? auditTitle.textContent : 'Audit title unvailable',
-        };
+    beforeEach(() => {
+      onApiRequestInterception = new Promise(resolve => apiRequestInterceptionResolve = resolve);
+    });
+
+    it('should call out to PSI with all categories by default', async () => {
+      const url = `${viewerUrl}?psiurl=https://www.example.com`;
+      await viewerPage.goto(url);
+
+      // Intercept and respond with sample lhr.
+      const interceptedRequest = await onApiRequestInterception;
+      const interceptedUrl = new URL(interceptedRequest.url());
+      expect(interceptedUrl.origin + interceptedUrl.pathname).toEqual('https://www.googleapis.com/pagespeedonline/v5/runPagespeed');
+      expect(interceptedUrl.searchParams.get('url')).toEqual('https://www.example.com');
+      expect(interceptedUrl.searchParams.getAll('category')).toEqual([
+        'performance',
+        'accessibility',
+        'seo',
+        'best-practices',
+        'pwa',
+      ]);
+
+      const psiResponse = {
+        lighthouseResult: JSON.parse(fs.readFileSync(sampleLhr, 'utf-8')),
+      };
+      await interceptedRequest.respond({
+        status: 200,
+        contentType: 'application/json',
+        body: JSON.stringify(psiResponse),
       });
-    }
 
-    const errorSelectors = '.lh-audit-explanation, .tooltip--error';
-    const auditErrors = await viewerPage.$$eval(errorSelectors, getErrors, selectors);
-    const errors = auditErrors.filter(item => item.explanation.includes('Audit error:'));
-    assert.deepStrictEqual(errors, [], 'Audit errors found within the report');
+      // Wait for report to render.
+      await viewerPage.waitForSelector('.lh-columns');
+
+      // No errors.
+      assert.deepStrictEqual(pageErrors, []);
+
+      // All categories.
+      const categories = await getCategoryElementsIds();
+      assert.deepStrictEqual(
+        categories.sort(),
+        lighthouseCategories.sort(),
+        `all categories not found`
+      );
+
+      // Should not clear the query string.
+      expect(await viewerPage.url()).toEqual(url);
+    });
+
+    it('should call out to PSI with specified categoeries', async () => {
+      const url = `${viewerUrl}?psiurl=https://www.example.com&category=seo&category=pwa`;
+      await viewerPage.goto(url);
+
+      const interceptedRequest = await onApiRequestInterception;
+      const interceptedUrl = new URL(interceptedRequest.url());
+      expect(interceptedUrl.origin + interceptedUrl.pathname).toEqual('https://www.googleapis.com/pagespeedonline/v5/runPagespeed');
+      expect(interceptedUrl.searchParams.get('url')).toEqual('https://www.example.com');
+      expect(interceptedUrl.searchParams.getAll('category')).toEqual([
+        'seo',
+        'pwa',
+      ]);
```

Comment:
for these, maybe cleaner to do `expect([... interceptedUrl.searchParams.entries()]).toEqual([...])`?

Reply:
I think these query params should be tested one at a time.

Reply:
>> for these, maybe cleaner to do `expect([... interceptedUrl.searchParams.entries()]).toEqual([...])`?

> I think these query params should be tested one at a time.

that's fine, I guess what I was really suggesting is that we should be testing all of them (especially `strategy` since it gets a default? and probably key, though the exact value matters less) and that form was just a way to cut down on the `expect(...).toEqual(...)` verboseness

Reply:
OK - done